### DISCUSSION TITLE
So how do I match beginning of a file?

### DISCUSSION THREAD

**Discussion by LD-72** [dnGrep/dnGrep#599] [Q&A] [answered]
"\A" and "^" both match the beginning of a line. The help page suggests that this is optional and they can match the beginning of file instead, but I can't find a setting for it. I also tried "^^" which works in another app I use, but that also seems to match the beginning of each line. How do I match only at the beginning of the file?

**Comment by doug24** [comment, marked as answer]
You can use `\A` and the `Multiline` option.  Without the Multiline option, dnGrep processes one line at a time and does not do any special handling for the position in the file.  With the Multiline option, the whole file is read before processing. 

From the [Microsoft documentation](https://docs.microsoft.com/en-us/dotnet/standard/base-types/anchors-in-regular-expressions#start-of-string-only-a) on .NET Regular Expressions:
>The \A anchor specifies that a match must occur at the beginning of the input string. It is identical to the ^ anchor, except that \A ignores the RegexOptions.Multiline option. Therefore, it can only match the start of the first line in a multiline input string.

Without Multiline:
<img src="https://user-images.githubusercontent.com/17227248/146624423-94ec9d27-5645-4db4-bc9d-c872005b211f.png" width="440"/>

With Multiline:
<img src="https://user-images.githubusercontent.com/17227248/146624482-bb4839bc-fb7b-4a30-b3d6-3d34458490f8.png" width="440"/>
